src/Code/index.js

```diff
@@ -0,0 +1,229 @@
+  textarea.current.select()
+  document.execCommand('copy')
+  openTooltip(setIsTooltipOpen)
+}
+
+const parseMetastring = (metastring) => {
+  const matches = metastring.match(/{.*?(?=})}/g)
+  const options = {}
+  if (matches) {
+    for (let i = 0; i < matches.length; i++) {
+      const option = matches[i].slice(1, -1).split(':')
+      const value = option[1] ? options[1].trim() : ''
+      if (value.toLowerCase() === 'true') {
+        options[option[0]] = true
+      } else if (value.toLowerCase() === 'false') {
+        options[option[0]] = false
+      } else {
+        options[option[0]] = value
+      }
+    }
+  }
+  return options
+}
+
+const RE = /{([\d,-]+)}/
+
+const calculateLinesToHighlight = (meta) => {
+  if (!RE.test(meta)) {
+    return () => false
+  }
+  const lineNumbers = RE.exec(meta)[1]
+    .split(`,`)
+    .map((v) => v.split(`-`).map((x) => parseInt(x, 10)))
+  console.log(lineNumbers)
+
+  return (index) => {
+    const lineNumber = index + 1
+    const inRange = lineNumbers.some(([start, end]) =>
+      end ? lineNumber >= start && lineNumber <= end : lineNumber === start
+    )
+    return inRange
+  }
+}
+
+const Code = ({
+  children,
+  className = '',
+  theme = 'dark',
+  copyButton = true,
+  lineNumbers = true,
+  metastring = '',
+  ...props
+}) => {
+  const [isTooltipOpen, setIsTooltipOpen] = useState(false)
+  const language = className.replace(/language-/, '')
+
+  const shouldHighlightLine = calculateLinesToHighlight(metastring)
+  const options = parseMetastring(metastring)
+  const isCopyButton = options.copy !== undefined ? options.copy : copyButton
+  const isLineNumbers =
+    options.numberLines !== undefined ? options.numberLines : lineNumbers
+
+  return (
+    <Provider
+      theme={defaultTheme}
+      colorScheme={theme}
+      scale='medium'
+      UNSAFE_style={{
+        borderRadius: 'var(--spectrum-global-dimension-size-50)'
+      }}
+      {...props}
+    >
+      <Highlight {...defaultProps} code={children} language={language}>
+        {({ className, tokens, getLineProps, getTokenProps }) => {
+          const lines = tokens.slice(0, -1)
+          const isMultiLine = lines.length > 1
+          const textarea = createRef()
+
+          return (
+            <pre className={classNames(className, 'spectrum-Code4')}>
+              {isCopyButton && (
+                <div
+                  css={css`
+                    position: relative;
+                  `}
+                >
+                  <textarea
+                    readOnly
+                    aria-hidden='true'
+                    css={css`
+                      position: fixed;
+                      left: -999px;
+                      opacity: 0;
+                    `}
+                    ref={textarea}
+                    value={children}
+                  />
+                  <ActionButton
+                    aria-describedby={tooltipId}
+                    onClick={() => {
```

Comment:
it should be `onPress` and you can wrap the ActionButton inside a View since it supports setting absolute positioning.

Reply:
Switched to onPress. For positioning I'm going to leave it for now. I may have to refactor this to allow the code to wrap around the button anyway.